numpy/_typing/__init__.py

```diff
@@ -2,149 +2,384 @@
 
 from __future__ import annotations
 
-from ._nested_sequence import (
-    _NestedSequence as _NestedSequence,
+from ._array_like import (
+    ArrayLike as ArrayLike,
 )
-from ._nbit_base import (
-    NBitBase as NBitBase,  # pyright: ignore[reportDeprecated]
-    _8Bit as _8Bit,
-    _16Bit as _16Bit,
-    _32Bit as _32Bit,
-    _64Bit as _64Bit,
-    _96Bit as _96Bit,
-    _128Bit as _128Bit,
+from ._array_like import (
+    NDArray as NDArray,
 )
-from ._nbit import (
-    _NBitByte as _NBitByte,
-    _NBitShort as _NBitShort,
-    _NBitIntC as _NBitIntC,
-    _NBitIntP as _NBitIntP,
-    _NBitInt as _NBitInt,
-    _NBitLong as _NBitLong,
-    _NBitLongLong as _NBitLongLong,
-    _NBitHalf as _NBitHalf,
-    _NBitSingle as _NBitSingle,
-    _NBitDouble as _NBitDouble,
-    _NBitLongDouble as _NBitLongDouble,
+from ._array_like import (
```

Comment:
I'm curious about the logic for this reformatting. Why the separation and  parenthesis? It doesn't seem to apply when the names aren't prefixed with `_`.

Reply:
See [unused-import (F401)](https://docs.astral.sh/ruff/rules/unused-import/#unused-import-f401):
> If an import statement is used to re-export a symbol as part of a module's public interface, consider using a "redundant" import alias, which instructs Ruff (and other tools) to respect the re-export, and avoid marking it as unused, as in:
> ```python
> from module import member as member
> ```
> Alternatively, you can use `__all__` to declare a symbol as part of the module's interface, as in:
> ```python
> # __init__.py
> import some_module
> 
> __all__ = ["some_module"]
> ```